galsim/phase_psf.py

```diff
@@ -1085,6 +1094,14 @@ def _finalize(self, flux, suppress_warning):
         self.img *= (flux / (self.img.sum() * self.scale**2))
         self.img = galsim.ImageD(self.img.astype(np.float64), scale=self.scale)
 
+        if self.max_size is not None and self.max_size < self.img.scale*self.img.array.shape[0]:
+            center = self.img.center()
+            bds = galsim.BoundsI(center)
+            half_max_size = int(np.ceil(0.5*self.max_size/self.img.scale))
+            bds += center + galsim.PositionI(-half_max_size, -half_max_size)
+            bds += center + galsim.PositionI(half_max_size, half_max_size)
+            self.img = self.img[bds]
```

Comment:
My recollection was that this was a substantial effect for the Great3 variable PSF sims, which we don't have currently in the examples/great3 directory.  But we can mock it up with cgc by re-making a PSF for each object with psf.index_key=obj_num.  Here are some timings.

On this branch:
```
$ galsim cgc.yaml output.nfiles=1 output.noclobber=False psf.index_key=obj_num image.nx_tiles=1
File 0 = control/ground/constant/image-000-0.fits: time = 30.303779 sec
$ galsim cgc.yaml output.nfiles=1 output.noclobber=False psf.index_key=obj_num image.nx_tiles=1 psf.items.1.max_size=1.e500
File 0 = control/ground/constant/image-000-0.fits: time = 30.807760 sec
```

On master:
```
$ galsim cgc.yaml output.nfiles=1 output.noclobber=False psf.index_key=obj_num image.nx_tiles=1
File 0 = control/ground/constant/image-000-0.fits: time = 23.059284 sec
$ galsim cgc.yaml output.nfiles=1 output.noclobber=False psf.index_key=obj_num image.nx_tiles=1 psf.items.1.max_size=1.e500
File 0 = control/ground/constant/image-000-0.fits: time = 29.484686 sec
```

So at least for the kind of application this parameter was designed for, the new implementation doesn't help.  The idea was really to speed up the PSF image generation process by not making as large an image.  

But if you think this is invalid, I'll take your word for it, since you understand this stuff much better than I do.  In that case, I'd recommend just deprecating max_size and have the message say that the previous use to speed up the calculation may lead to aliasing problems, so we removed it, but if you don't mind the aliasing, then you can speed up the calculation similarly by lowering pad_factor below the default 1.5.